### DISCUSSION TITLE
Is there a way to only do the npm run build to a single .svelte file?

### DISCUSSION THREAD

**Discussion by JuanjoJG-WATA** [open-webui/open-webui#2976] [Q&A]
Hello, I've been making adjustments to several files, and after each change, I find myself needing to execute 'npm run build', which processes every Svelte file. Is there a method to only compile the file I've modified? For instance, if I make alterations to 'General.svelte', I'd prefer not to compile every Svelte file, which currently takes about 2 minutes with each change. Any suggestions on how to achieve this?

**Comment by jabbasj** [comment]
Hey I too would like to know the answer to your question.
Btw are you executing this "npm run build" command locally and mapping a volume with the build files to the container ? 

This feels finicky, I wonder what is the 'clean' way to modify the front-end while developing (& using docker)

**Reply by JuanjoJG-WATA** [reply]
Sadly, I'm not using Docker. I used to, but I found it hard to modify to my needs, so I used the manual installation, using Node and Python.

What I do is, I modify any .svelte file with any change I want, using Visual Studio Code for example, and then I just run npm run build, but that affects to every svelte file, not just the one I modified which would be way faster.

It doesn't seem to be the most clean to do it to me neither, but it's the easiest one I've found. Before this I was modifying the .js files that are created after building the front end.

**Reply by jabbasj** [reply]
Ok so if you're doing this for development without Docker, I found a way that should work for you:

1) run "npm run dev" instead of "npm run build"
2) run the backend with the bash script or "uvicorn main:app --host 0.0.0.0 --port 8080 --forwarded-allow-ips '*' --reload"
3) now access the front-end using the dev url/port (default is localhost:5173)

Now you should see updates to your svelte files immediately in your browser
